windows/ReactNativeWebView2/ReactWebView2Manager.cpp

```diff
@@ -0,0 +1,145 @@
+
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+    using namespace Windows::Foundation::Collections;
+    using namespace Windows::UI;
+    using namespace Windows::UI::Xaml;
+    using namespace Windows::UI::Xaml::Controls;
+    using namespace Microsoft::UI::Xaml::Controls;
+    using namespace Windows::Web::Http;
+    using namespace Windows::Web::Http::Headers;
+}
+
+namespace winrt::ReactNativeWebView2::implementation {
+
+    ReactWebView2Manager::ReactWebView2Manager() {}
+
+    // IViewManager
+    winrt::hstring ReactWebView2Manager::Name() noexcept {
+        return L"RCTWebView2";
+    }
+
+    winrt::FrameworkElement ReactWebView2Manager::CreateView() noexcept {
+      auto view = winrt::ReactNativeWebView2::ReactWebView2(m_reactContext);
+      return view;
+    }
+
+    // IViewManagerWithReactContext
+    winrt::IReactContext ReactWebView2Manager::ReactContext() noexcept {
+        return m_reactContext;
+    }
+
+    void ReactWebView2Manager::ReactContext(IReactContext reactContext) noexcept {
+        m_reactContext = reactContext;
+    }
+
+    // IViewManagerWithNativeProperties
+    IMapView<hstring, ViewManagerPropertyType> ReactWebView2Manager::NativeProps() noexcept {
+        auto nativeProps = winrt::single_threaded_map<hstring, ViewManagerPropertyType>();
+        nativeProps.Insert(L"source", ViewManagerPropertyType::Map);
+        return nativeProps.GetView();
+    }
+
+    void ReactWebView2Manager::UpdateProperties(
+        FrameworkElement const& view,
+        IJSValueReader const& propertyMapReader) noexcept {
+        auto control = view.as<winrt::UserControl>();
+        auto content = control.Content();
+        auto webView = content.as<winrt::WebView2>();
+        const JSValueObject& propertyMap = JSValueObject::ReadFrom(propertyMapReader);
+
+        for (auto const& pair : propertyMap) {
+            auto const& propertyName = pair.first;
+            auto const& propertyValue = pair.second;
+            if (propertyValue.IsNull()) continue;
+
+            if (propertyName == "source") {
+                auto const& srcMap = propertyValue.AsObject();
+                if (srcMap.find("uri") != srcMap.end()) {
+                    auto uriString = srcMap.at("uri").AsString();
+                    if (uriString.length() == 0) {
+                        continue;
+                    }
+
+                    bool isPackagerAsset = false;
+                    if (srcMap.find("__packager_asset") != srcMap.end()) {
+                        isPackagerAsset = srcMap.at("__packager_asset").AsBoolean();
+                    }
+                    if (isPackagerAsset && uriString.find("file://") == 0) {
+                        auto bundleRootPath = winrt::to_string(ReactNativeHost().InstanceSettings().BundleRootPath());
+                        uriString.replace(0, 7, bundleRootPath.empty() ? "ms-appx-web:///Bundle/" : bundleRootPath);
```

Comment:
consider replacing the 7 with std::size("file://"), maybe have that string defined somewhere to avoid repeating it (you can make it constexpr)